src/lib.rs

```diff
@@ -72,27 +70,32 @@ impl PoseidonHasher {
         PoseidonHash::hash_no_pad(&x).to_bytes()
     }
 
-
     // This function should only be used to compute the quantus storage key for Transfer Proofs
     // It breaks up the bytes input in a specific way that mimics how our zk-circuit does it
-    pub fn hash_storage(x: &[u8]) -> [u8; 32] {
-        const EXPECTED_STORAGE_PREIMAGE_LEN: usize = 4 + 32 + 32 + 16;
-        debug_assert!(x.len() == EXPECTED_STORAGE_PREIMAGE_LEN, "Input must be exactly 84 bytes");
-        let mut felts = Vec::with_capacity(EXPECTED_STORAGE_PREIMAGE_LEN);
-        type AccountId = [u8; 32];
+    pub fn hash_storage<AccountId: Decode + Encode + MaxEncodedLen>(x: &[u8]) -> [u8; 32] {
+        let expected_storage_len = u32::max_encoded_len()
+            + AccountId::max_encoded_len()
+            + AccountId::max_encoded_len()
+            + u128::max_encoded_len();
+        debug_assert!(
+            x.len() == expected_storage_len,
+            "Input must be exactly 84 bytes"
+        );
+        let mut felts = Vec::with_capacity(expected_storage_len);
         let mut y = x;
         let (nonce, from_account, to_account, amount): (u32, AccountId, AccountId, u128) =
```

Comment:
Is there a good reason we pass in a generic u8 byte array instead of passing in all these nice objects we expect the byte array to contain as parameters?

Reply:
I guess I did not understand this PR before as I thought it was for any u8 array

Reply:
yes because the trait is defined that way in another package